test/inc/zoo/variant.h

```diff
@@ -73,8 +90,24 @@ template<typename Visitor, typename Var>
 auto visit(Visitor &&visitor, Var &&var) ->
     decltype(visitor(get<0>(std::forward<Var>(var))));
 
+namespace detail {
+
+    template<typename T, typename Enable = void>
+    struct DestroyInvoker {
+    };
+
+    template<template<typename...> class Var, typename... Ts>
+    struct DestroyInvoker<Var<Ts...>,
+            std::enable_if_t<!(std::is_trivially_destructible_v<Ts> && ...)>> {
+        ~DestroyInvoker() {
+            static_cast<Var<Ts...>*>(this)->destroy();
```

Comment:
If all this does is to potentially invoke `destroy`, `destroy` could be made a template with `enable_if`

Reply:
Nope, you need to actually remove the destructor, otherwise a defined destructor (even if it only contains an "if constexpr" that calls destroy will make it non-trivially destructible).

Reply:
No disagreement, I meant in the non-trivial case you will have an explicit destructor, in the trivial, an implicit or defaulted destructor, thus this is not necessary

Reply:
No disagreement, I meant in the non-trivial case you will have an explicit destructor, in the trivial, an implicit or defaulted destructor, thus this is not necessary